### ISSUE TITLE
opentelemetry-collector.convertMemToMib produces 0 if using a decimal point

### ISSUE THREAD

**Issue by snuggie12** [open-telemetry/opentelemetry-helm-charts#1006]
This is fairly easily worked around, but it did cause our HPAs to max out so figured I'd make the ticket in case anyone else runs into it.

We had a memory limit of `1.5Gi` in place and the `atoi` function will convert that to `0`. This is especially bad with `GOMEMLIMIT` since it out of nowhere started using 100% of the CPU.

**Comment by TylerHelmuth** [comment]
You're not wrong, but also I don't have a great solution in mind for this other than don't use decimals. Maybe https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30289 can help
